src/java/org/apache/cassandra/schema/ColumnMetadata.java

```diff
@@ -126,42 +135,60 @@ private static long comparisonOrder(Kind kind, boolean isComplex, long position,
     {
-        return new ColumnMetadata(table, name, type, position, Kind.PARTITION_KEY, null);
+        return new ColumnMetadata(table, name, type, position, Kind.PARTITION_KEY, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata partitionKeyColumn(String keyspace, String table, String name, AbstractType<?> type, int position)
     {
-        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, position, Kind.PARTITION_KEY, null);
+        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, position, Kind.PARTITION_KEY, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata clusteringColumn(TableMetadata table, ByteBuffer name, AbstractType<?> type, int position)
     {
-        return new ColumnMetadata(table, name, type, position, Kind.CLUSTERING, null);
+        return new ColumnMetadata(table, name, type, position, Kind.CLUSTERING, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata clusteringColumn(String keyspace, String table, String name, AbstractType<?> type, int position)
     {
-        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, position, Kind.CLUSTERING, null);
+        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, position, Kind.CLUSTERING, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata regularColumn(TableMetadata table, ByteBuffer name, AbstractType<?> type)
     {
-        return new ColumnMetadata(table, name, type, NO_POSITION, Kind.REGULAR, null);
+        return new ColumnMetadata(table, name, type, NO_POSITION, Kind.REGULAR, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata regularColumn(String keyspace, String table, String name, AbstractType<?> type)
     {
-        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, NO_POSITION, Kind.REGULAR, null);
+        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, NO_POSITION, Kind.REGULAR, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata staticColumn(TableMetadata table, ByteBuffer name, AbstractType<?> type)
     {
-        return new ColumnMetadata(table, name, type, NO_POSITION, Kind.STATIC, null);
+        return new ColumnMetadata(table, name, type, NO_POSITION, Kind.STATIC, null, ColumnConstraints.NO_OP);
     }
 
     public static ColumnMetadata staticColumn(String keyspace, String table, String name, AbstractType<?> type)
     {
-        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, NO_POSITION, Kind.STATIC, null);
+        return new ColumnMetadata(keyspace, table, ColumnIdentifier.getInterned(name, true), type, NO_POSITION, Kind.STATIC, null, ColumnConstraints.NO_OP);
+    }
+
+    public ColumnMetadata(TableMetadata table,
```

Comment:
Stylistic idea here. Right now you've added `ColumnConstraints.NO_OP` to all the method calls here which is a lot of noise with very little signal. What if we had:
```java
static ColumnMetadata buildWithoutConstraints(...) {}
static ColumnMetadata buildWithConstraints(..., @Nonnull ColumnConstraints columnConstraints){]
```
Where the 1st method we'd call w/all the callers in this class so we don't have to pass that redundant `NO_OP` in all of them, and we rely on method names to help disambiguate what kind of construction we're doing?

Don't feel super strongly on this but got similar feedback from David a couple weeks ago on this w/callbacks and I liked the aesthetics of how the end result turned out.

Reply:
I would have agreed with you, but in this particular case, same patter is used with other of the params (`mask`). I don't think we can come with different names for the different dimensions we would have to deal with, so my take would be to leave it as is for this instance. This is something that always can be refactored if more parameters keep being added.

Reply:
Yeah, makes sense. Not a blocker and matrixing of multiple things that have this optionality lead directly into builder and constructor chaining hell. I don't think this area is bad enough to warrant any of that so we can just go with this.